### PR TITLE
Fix License-Violations, Add Javadoc, Fix Temperature Modifier and wor…

### PR DESCRIPTION
**Pull request by shartte**
…k on public API (getters)

### CODE REVIEW

**Comment by shartte** [comment]
This is not done yet though.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java [hunk 1]
```diff
@@ -207,81 +229,315 @@ public Biome build() {
...
+	/**
+	 * Checks whether this builder contains a given feature (in any configuration) in any generation step.
+	 *
+	 * @see Biome#getGenerationSettings()
+	 * @see GenerationSettings#getFeatures()
+	 */
+	public boolean hasFeature(Feature<?> feature) {
+		for (List<Supplier<ConfiguredFeature<?, ?>>> featureList : features.values()) {
+			for (Supplier<ConfiguredFeature<?, ?>> supplier : featureList) {
+				if (supplier.get().feature == feature) {
```
Comment (CheaterCodes) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java:386]: supplier.get() might still be null at this point

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java [hunk 2]
```diff
@@ -207,81 +229,315 @@ public Biome build() {
...
+	 *
+	 * @see Biome#getGenerationSettings()
+	 * @see GenerationSettings#getFeatures()
+	 */
+	public boolean hasFeature(GenerationStep.Feature step, Feature<?> feature) {
+		List<Supplier<ConfiguredFeature<?, ?>>> featureList = features.get(step);
+
+		if (featureList != null) {
+			for (Supplier<ConfiguredFeature<?, ?>> supplier : featureList) {
+				if (supplier.get().feature == feature) {
```
Comment (CheaterCodes) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java:407]: supplier.get() might still be null at this point.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java [hunk 3]
```diff
@@ -207,81 +229,315 @@ public Biome build() {
...
+	 * @see Biome#getGenerationSettings()
+	 * @see GenerationSettings.Builder#structureFeature(ConfiguredStructureFeature)
+	 */
 	public FabricBiomeBuilder structureFeature(RegistryKey<ConfiguredStructureFeature<?, ?>> structureFeature) {
-		structureFeature(() -> registryManager.get(Registry.CONFIGURED_STRUCTURE_FEATURE_WORLDGEN).get(structureFeature));
+		this.structureFeatures.add(() -> configuredStructureFeaturesRegistry.get(structureFeature));
 		return this;
 	}
 
-	public FabricBiomeBuilder surfaceBuilder(Supplier<ConfiguredSurfaceBuilder<?>> surfaceBuilder) {
```
Comment (CheaterCodes) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java:275]: I don't see a big issue with this staying public, I'd rather keep it public for the few people who might want it.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java [hunk 4]
```diff
@@ -207,81 +229,315 @@ public Biome build() {
...
+	/**
+	 * @see Biome#getGenerationSettings()
+	 * @see GenerationSettings.Builder#carver(GenerationStep.Carver, ConfiguredCarver)
+	 */
 	public FabricBiomeBuilder carver(GenerationStep.Carver step, RegistryKey<ConfiguredCarver<?>> carver) {
 		carver(step, () -> registryManager.get(Registry.CONFIGURED_CARVER_WORLDGEN).get(carver));
 		return this;
 	}
 
-	public FabricBiomeBuilder feature(GenerationStep.Feature step, Supplier<ConfiguredFeature<?, ?>> feature) {
```
Comment (CheaterCodes) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java:255]: I don't see a big issue with this staying public, I'd rather keep it public for the few people who might want it.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java [hunk 5]
```diff
@@ -207,81 +229,315 @@ public Biome build() {
...
+	/**
+	 * @see Biome#getGenerationSettings()
+	 * @see GenerationSettings.Builder#feature(GenerationStep.Feature, ConfiguredFeature)
+	 */
 	public FabricBiomeBuilder feature(GenerationStep.Feature step, RegistryKey<ConfiguredFeature<?, ?>> feature) {
 		feature(step, () -> registryManager.get(Registry.CONFIGURED_FEATURE_WORLDGEN).get(feature));
 		return this;
 	}
 
-	public FabricBiomeBuilder structureFeature(Supplier<ConfiguredStructureFeature<?, ?>> structureFeature) {
```
Comment (CheaterCodes) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java:265]: I don't see a big issue with this staying public, I'd rather keep it public for the few people who might want it.

File: fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java [hunk 6]
```diff
@@ -207,81 +229,315 @@ public Biome build() {
...
+
+	/**
+	 * @see Biome.Builder#temperatureModifier(Biome.TemperatureModifier)
+	 */
 	public FabricBiomeBuilder temperatureModifier(Biome.TemperatureModifier temperatureModifier) {
 		this.temperatureModifier = temperatureModifier;
 		return this;
 	}
 
-	public FabricBiomeBuilder carver(GenerationStep.Carver step, Supplier<ConfiguredCarver<?>> carver) {
```
Comment (CheaterCodes) [inline fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/FabricBiomeBuilder.java:245]: I don't see a big issue with this staying public, I'd rather keep it public for the few people who might want it.

**Review by CheaterCodes — CHANGES_REQUESTED** [review]
Thanks for helping out with this!
I'd rather keep the suppliers public, I don't think they can do much harm.
Other than that, I'll merge once that's changed.
If you want to chenge them to private, let's discuss that in the actual PR.

**Pull request closed without merging**